pkg/aws/ec2/ec2.go

```diff
@@ -691,6 +694,51 @@ func (c *Client) UnassignENIPrefixes(ctx context.Context, eniID string, prefixes
 	return err
 }
 
+func (c *Client) AssociateEIP(ctx context.Context, instanceID string, eipTags ipamTypes.Tags) (string, error) {
+	filters := make([]ec2_types.Filter, 0, len(eipTags))
+	for k, v := range eipTags {
+		filters = append(filters, ec2_types.Filter{
+			Name:   aws.String(fmt.Sprintf("tag:%s", k)),
+			Values: []string{v},
+		})
+	}
+
+	describeAddressesInput := &ec2.DescribeAddressesInput{
+		Filters: filters,
+	}
+	c.limiter.Limit(ctx, "DescribeAddresses")
+	sinceStart := spanstat.Start()
+	addresses, err := c.ec2Client.DescribeAddresses(ctx, describeAddressesInput)
+	c.metricsAPI.ObserveAPICall("DescribeAddresses", deriveStatus(err), sinceStart.Seconds())
+	if err != nil {
+		return "", err
+	}
+	log.Infof("Found %d EIPs corresponding to tags %v", len(addresses.Addresses), eipTags)
+
+	for _, address := range addresses.Addresses {
+		// Only pick unassociated EIPs
+		if address.AssociationId == nil {
+			associateAddressInput := &ec2.AssociateAddressInput{
+				AllocationId:       address.AllocationId,
+				AllowReassociation: aws.Bool(false),
+				InstanceId:         aws.String(instanceID),
+			}
+			c.limiter.Limit(ctx, "AssociateAddress")
+			sinceStart = spanstat.Start()
+			association, err := c.ec2Client.AssociateAddress(ctx, associateAddressInput)
+			c.metricsAPI.ObserveAPICall("AssociateAddress", deriveStatus(err), sinceStart.Seconds())
+			if err != nil {
+				// TODO some errors can probably be skipped and next EIP can be tried
```

Comment:
> What if we did a continue and attempt the next one in this iteration itself ?

That's exactly what I was thinking about, basically doing a `continue` and trying to associate the next EIP. Some association errors can definitely be benign, for example if there is a race condition and the EIP is assigned by someone else to another instance between the `DescribeAddresses` and the `AssociateAddress` calls made here. 

However, I am not sure whether all errors can be / should be skipped? One potential failure scenario I can see is with the AWS API returning an error but the association actually succeeding. So when we try to associate the next EIP, we'll get an error saying that the instance already has an associated EIP (because `AllowReassociation` is `false`). So we'll need some logic to short-circuit the retries in that case. That's just one scenario I thought about, not sure if there's anything else we need to handle

Reply:
Alright, so, if we're thinking the next reconciliation loop will automatically handle some failure cases, I think that's fine. What is the min interval between two consecutive reconciliation loops ?

Reply:
The min interval is 10ms: https://github.com/cilium/cilium/blob/fe7b5e8a94a7fa39aa80447013911ddc17b2cb77/pkg/ipam/node_manager.go#L324